node/node.go

```diff
@@ -142,6 +142,10 @@ type Node struct {
 	// This exists in addition to lastNodeRole to support older CAs that
 	// only fill in the DesiredRole field.
 	lastNodeDesiredRole lastSeenRole
+	// expectingRoleChange is set if DesiredRole and Role do not match,
+	// meaning the Role update may come later (if the CA is recent enough
+	// to use Role).
+	expectingRoleChange bool
```

Comment:
I can actually replace this with something like:

```go
*n.lastNodeDesiredRole.role != *n.lastNodeRole.role
```

evaluated before calling `observe`... if people think that's cleaner. I guess keeping less state around is nice.

I can also remove the pointer from `lastSeenRole` now that it is always initialized.